pkg/telemetry/metadata.go

```diff
@@ -0,0 +1,71 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package telemetry
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"strings"
+	"syscall"
+
+	"github.com/timescale/promscale/pkg/version"
+)
+
+type (
+	Metadata map[string]string
+	Stats    map[string]string
+)
+
+func promscaleMetadata() (Metadata, error) {
+	metadata := Metadata{
+		"promscale_version":     version.Promscale,
+		"promscale_commit_hash": version.CommitHash,
+		"promscale_arch":        runtime.GOARCH,
+		"promscale_os":          runtime.GOOS,
+	}
+	uname := syscall.Utsname{}
+	if err := syscall.Uname(&uname); err != nil {
+		return metadata, fmt.Errorf("syscall uname: %w", err)
+	}
+	metadata["promscale_os_sys_name"] = toString(uname.Sysname)
+	metadata["promscale_os_node_name"] = toString(uname.Nodename)
+	metadata["promscale_os_release"] = toString(uname.Release)
+	metadata["promscale_os_version"] = toString(uname.Version)
+	metadata["promscale_os_machine"] = toString(uname.Machine)
+	metadata["promscale_os_domain_name"] = toString(uname.Domainname)
+	return metadata, nil
+}
+
+func toString(prop [65]int8) string {
```

Comment:
why dont we use the method in https://gitlab.cncf.ci/prometheus/prometheus/blob/3128875ff4de14527502be787e3f3eb480a11fc9/pkg/runtime/uname_linux.go#L30

`return string(prop[:bytes.IndexByte(prop[:], 0)])`

Reply:
That is `unix.Utsname{}`, which has `[65]byte` as type against the `syscall.Utsname{}` which has `[65]int8`. We cannot directly use `bytes.IndexByte()` with a `int8` without first converting into a slice of `byte`, which make the current approach better. Plus I don't think we should be using `unix` as windows systems would not be happy.